addons/stock/models/stock_picking.py

```diff
@@ -542,6 +542,7 @@ def _search_delay_alert_date(self, operator, value):
     @api.onchange('picking_type_id', 'partner_id')
     def onchange_picking_type(self):
         if self.picking_type_id:
+            self = self.with_company(self.company_id)
```

Comment:
the company_id of a picking is a stored related to the company of the picking type, it seems to me this should be 
``` python
self = self.with_company(self.picking_type_id.company_id)
```

Reply:
The compute of the related field should be triggered when reading it, but I can change it if you find it safer.

Reply:
ok just tested it and indeed self.company_id is the new one
i think it's better to be explicit, but up to you